integration-tests/utils/integration-utils.js

```diff
@@ -7,6 +7,27 @@ const { screenshotDirPath } = require('./integration-config');
 
 const defaultToVisibleTrue = _.defaults({ visible: true });
 
+/**
+ * Repeat a given function until it evaluates to truthy, returning the final result. Iterates synchronously.
+ * @param getResult the function to repeat, returning a truthy or falsy value
+ * @param interval the number of milliseconds to wait between attempts
+ * @leading whether to check getResult before the initial interval wait
+ * @retries how many times to retry before aborting
+ */
+const retryUntil = async ({ getResult, interval = Millis.ofSecond, leading = false, retries = 5 }) => {
```

Comment:
How does this differ from `waitForFn`?

Reply:
This method uses a synchronous do-while loop. waitForFn is using setInterval and in my tests I was seeing it not iterating past the first attempt - I wasnt sure if it was due to an issue with timing (running slow UI flows possibly overlapping) or context (losing `page` or `this`, maybe?) I tried for awhile then settled for factoring the common logic into a single helper.